tinygrad/runtime/ops_kfd.py

```diff
@@ -17,7 +17,14 @@
 libc.munmap.argtypes = [ctypes.c_void_p, ctypes.c_size_t]
 libc.munmap.restype = ctypes.c_int
 
-def node_sysfs_path(node_id, file): return f"/sys/devices/virtual/kfd/kfd/topology/nodes/{node_id}/{file}"
+def is_usable_gpu(gpu_id):
+    try:
+      with gpu_id.open() as f:
+        return int(f.read()) != 0
+    except OSError:
+      return False
+gpus = [g.parent for g in pathlib.Path("/sys/devices/virtual/kfd/kfd/topology/nodes").glob("*/gpu_id") if is_usable_gpu(g)]
```

Comment:
This should be in the first call to KFDDevice __init__ (with `if KFDDevice.kfd == -1:`) so it doesn't run on import.